tests/integ/test_packaging.py

```diff
@@ -539,3 +575,255 @@ def plus_one_month(x):
     Utils.check_answer(
         df.select(plus_one_month_udf("a")).collect(), [Row(plus_one_month(d))]
     )
+
+
+@pytest.mark.skipif(
+    IS_IN_STORED_PROC,
+    reason="numpy and pandas are required",
+)
+def test_add_requirements_with_empty_stage_as_persist_path(
+    session, resources_path, temporary_stage
+):
+    """
+    Assert that adding a persist_path (empty stage) does not affect the requirements addition process.
+    """
+    test_files = TestFiles(resources_path)
+
+    session.add_requirements(
+        test_files.test_requirements_file, persist_path=temporary_stage
+    )
+    assert session.get_packages() == {
+        "numpy": "numpy==1.23.5",
+        "pandas": "pandas==1.5.3",
+    }
+
+    udf_name = Utils.random_name_for_temp_object(TempObjectType.FUNCTION)
+
+    @udf(name=udf_name, packages=["snowflake-snowpark-python==1.3.0"])
+    def get_numpy_pandas_version() -> str:
+        import snowflake.snowpark as snowpark
+
+        return f"{snowpark.__version__}"
+
+    Utils.check_answer(session.sql(f"select {udf_name}()"), [Row("1.3.0")])
+
+
+@pytest.mark.skipif(
+    IS_IN_STORED_PROC,
+    reason="subprocess calls are not possible within a stored procedure",
+)
+def test_add_requirements_unsupported_with_empty_stage_as_persist_path(
+    session, resources_path, temporary_stage
+):
+    """
+    Assert that adding a persist_path (empty stage) does not affect the requirements addition process, even if
+    requirements are unsupported.
+    """
+    test_files = TestFiles(resources_path)
+
+    with patch.object(session, "_is_anaconda_terms_acknowledged", lambda: True):
+        session.add_requirements(
+            test_files.test_unsupported_requirements_file, persist_path=temporary_stage
+        )
+
+    assert session.get_packages().keys() == {"matplotlib", "numpy", "pyyaml", "scipy"}
+
+    udf_name = Utils.random_name_for_temp_object(TempObjectType.FUNCTION)
+
+    @udf(name=udf_name)
+    def get_numpy_pandas_version() -> str:
+        import skfuzzy as fuzz
+
+        return fuzz.__version__
+
+    Utils.check_answer(session.sql(f"select {udf_name}()"), [Row("0.4.2")])
+
+
+@pytest.mark.skipif(
+    IS_IN_STORED_PROC,
+    reason="subprocess calls are not possible within a stored procedure",
+)
+def test_add_requirements_unsupported_with_persist_path_negative(
+    session, resources_path, temporary_stage
+):
+    """
+    Assert that failure in loading environment from `persist_path` does not affect the requirements addition process,
+    even if requirements are unsupported.
+    """
+    test_files = TestFiles(resources_path)
+
+    with patch.object(session, "_is_anaconda_terms_acknowledged", lambda: True):
+        with patch(
+            "snowflake.snowpark.session.Session._load_unsupported_packages_from_stage",
+            side_effect=Exception("This function does not work"),
+        ):
```

Comment:
Here, we force load from stage to fail, using patch.

Reply:
this is not really a negative test then, is it? since we don't expect `add_requirements` to fail

Reply:
true, changed the name of this test